esphome/components/mqtt/__init__.py

```diff
@@ -199,6 +201,8 @@ def validate_fingerprint(value):
             cv.Optional(CONF_CERTIFICATE_AUTHORITY): cv.All(
                 cv.string, cv.only_with_esp_idf
             ),
+            cv.Optional(CONF_CERTIFICATE): cv.All(cv.string, cv.only_with_esp_idf),
+            cv.Optional(CONF_KEY): cv.All(cv.string, cv.only_with_esp_idf),
```

Comment:
Since these are all or nothing, they should be marked as inclusive with each other
```suggestion
            cv.Inclusive(CONF_CERTIFICATE, "cert-key-pair"): cv.All(cv.string, cv.only_with_esp_idf),
            cv.Inclusive(CONF_KEY, "cert-key-pair"): cv.All(cv.string, cv.only_with_esp_idf),
```

Reply:
Should they also be named `CONF_CLIENT_CERTIFICATE` ?

Reply:
Also, arduino and esp-idf use the same mqtt backend so the validation should be `cv.only_on_esp32`

Reply:
> Since these are all or nothing, they should be marked as inclusive with each other

Done.

Reply:
> I didn't think you would want to add this to the global const table, but I will if you like.

Adding new consts is no problem. I would rather it be more descriptive of the actual configuration item

Reply:
Thanks, sorry i didn't get to it before you did :)